cli/module_build.go

```diff
@@ -876,6 +886,18 @@ func reloadModuleAction(c *cli.Context, vc *viamClient, args reloadModuleArgs, l
 	var partArch string
 	var platform string
 	if part.Part.UserSuppliedInfo != nil {
+		// Check if the viam-server version is supported for hot reloading
+		if part.Part.UserSuppliedInfo.Fields["version"] != nil {
+			version, err := semver.NewVersion(part.Part.UserSuppliedInfo.Fields["version"].GetStringValue())
+			if err != nil {
+				return fmt.Errorf("unable to determine viam-server version: %w", err)
```

Comment:
Can we add a carveout for `(dev)` version of CLI (which is what shows up for local builds)? This might occasionally be wrong but it seems better than erroring out when trying to do a hot reload from a test build.